### DISCUSSION TITLE
Lovelace entities slow to update?

### DISCUSSION THREAD

**Discussion by atv2016** [Koenkk/zigbee2mqtt#11336] [Devices]
Not sure, but if i physically turn off a light bulb the entity in lovelace doesn't seem to even understand that the light is off. With tuya, albeit slow to respond, would update.

Have it forgotten to configure something or is this the expected state ?

**Comment by atv2016** [comment]
@Koenkk Could you help me understand how quick this is supposed to update in lovelace? Much obliged.

**Comment by Koenkk** [comment]
> Not sure, but if i physically turn off a light bulb the entity in lovelace doesn't seem to even understand that the light is off.

The bulb doesn't send anything when it is removed from power, thus z2m doesn't know it is off.

You can use the [availability feature](https://www.zigbee2mqtt.io/guide/configuration/device-availability.html) to mark devices as unavailable, but this will not be done instantly (since z2m pings the device on a certain interval)

**Reply by atv2016** [reply]
Than @Koenkk. Does this also work for updating the entity when i switch a lightswitch to on ? I imagine if i set it to 5 pinging is done every 5 minutes, so it will be updated regardless if it's online or offline and update the entity accordingly?

I just set it to 1 minute, but it's reporting a whole bunch of plugs and lights that are definitely not offline..just getting a whole bunch of 'failed to ping'. 

Why does tuya seem to update almost instantly though? Not for turning lights of certainly, but when turning a lightswitch physically on, the entity gets almost updated instantly from what i've seen.

**Reply by Koenkk** [reply]
> Does this also work for updating the entity when i switch a lightswitch to on ?


When a bulb is powered on, it (most of them) sends a command so z2m immediately recognises it is on.

> Why does tuya seem to update almost instantly though?

TuYa device or when the bulb is paired to the TuYa bridge?
